plugins/out_clp_s3/flush/flush.go

```diff
@@ -0,0 +1,87 @@
+// Package implements method to send data to output
+package flush
+
+import (
+	"C"
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+	"unsafe"
+
+	"github.com/fluent/fluent-bit-go/output"
+
+	"github.com/y-scope/fluent-bit-clp/config"
+)
+
+// flushes data to file
+//
+// Parameters:
+//   - data: msgpack data
+//   - length: byte length
+//   - tag: fluent-bit tag
+//   - S3Config: configuration based on fluent-bit.conf
+//
+// Returns:
+//   - err: error flushing data
+func File(data unsafe.Pointer, length int, tag string, config *config.S3Config) error {
+	fullFilePath := filepath.Join(config.Path, config.File)
+
+	// If the file doesn't exist, create it, or append to the file
+	// will still cause error if there is no directory
+	f, err := os.OpenFile(fullFilePath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0o644)
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+
+	// ================== This code is mostly boilerplate [fluent-bit reference] ================== //
+	// temporary changes were made so that writes to file instead of stdout
+	// code will be deleted when switch to IR / send to S3
```

Comment:
format / line wrapping to 100

add TODO if this is to be removed (says it is temporary)